Signal/src/ViewControllers/InboxTableViewCell.m

```diff
@@ -227,16 +227,8 @@ - (void)configureWithThread:(TSThread *)thread
         self.unreadLabel.hidden = YES;
     }
 
-    NSString *threadIdCopy = thread.uniqueId;
-    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        UIImage *avatar =
-            [OWSAvatarBuilder buildImageForThread:thread contactsManager:contactsManager diameter:kavatarViewDiameter];
-        dispatch_async(dispatch_get_main_queue(), ^{
-            if ([_threadId isEqualToString:threadIdCopy]) {
-                self.avatarView.image = avatar;
-            }
-        });
-    });
+    self.avatarView.image =
```

Comment:
We initially talked about having this being async in the event of a cache miss, but in my testing, after fixing the caching, even on my iphone5 with 1k threads, this scrolls smoothly without egregious CPU (4% in generating avatars - when scrolling wildly, but it trends to zero when you are scrolling back and forth due to the cache hits).

I decided it wasn't worth the complexity.